pinot-druid/src/test/java/org/apache/pinot/druid/data/readers/DruidSegmentRecordReaderTest.java

```diff
@@ -0,0 +1,141 @@
+      Assert.assertTrue(row < records.length);
+
+      GenericRow nextRow = testReader.next();
+      GenericRow transformedRow = defaultTransformer.transform(nextRow);
+
+      for (int col = 0; col < columns.length; col++) {
+        if (records[row][col] != null) {
+          if (transformedRow.getValue(columns[col]) != null && records[row][col] != null) {
+            Assert.assertTrue(transformedRow.getValue(columns[col]).getClass() == records[row][col].getClass(),
+                "In row " + row + " and column " + col + ", type of value in segment (" + transformedRow.getValue(columns[col]).getClass()
+                    + ") does not match type of value in test record (" + records[row][col].getClass() + ").");
+            if (transformedRow.getValue(columns[col]).getClass() == String[].class) {
+              Assert.assertTrue(compareMultiValueColumn(transformedRow.getValue(columns[col]), records[row][col]));
+            } else {
+              Assert.assertEquals(transformedRow.getValue(columns[col]), records[row][col]);
+            }
+          }
+        }
+      }
+      genericRows.add(nextRow);
+      row += 1;
+    }
+    Assert.assertFalse(testReader.hasNext());
+    Assert.assertEquals(genericRows.size(), records.length, "Size of GenericRows (" + genericRows.size()
+        + ") must be equal to records size (" + records.length + ").");
+  }
+
+  private boolean compareMultiValueColumn(Object value1, Object value2) {
+    Object[] value1Array = (Object[]) value1;
+    Object[] value2Array = (Object[]) value2;
+    Set<Object> value1Set = new HashSet<>(Arrays.asList(value1Array));
+    Set<Object> value2Set = new HashSet<>(Arrays.asList(value2Array));
+    return value1Set.containsAll(value2Set);
+  }
+
+  @BeforeClass
+  public void setUp()
+      throws IOException {
+    // Wikipedia is sample data from Druid's Quickstart tutorial
+    wikipedia_test = createFileFromFilename("test_druid_wikipedia");
+    wikipedia_test_schema = createSchemaFromFilename("wikipedia-schema.json");
+
+    csv_test = createFileFromFilename("test_druid_from_csv");
+    csv_test_schema = createSchemaFromFilename("csv-test-schema.json");
+
+    all_types_test = createFileFromFilename("test-druid-all-types");
+    all_types_test_schema = createSchemaFromFilename("test-druid-all-types-schema.json");
+  }
+
+  @Test
+  public void testReadAllRows()
+      throws IOException {
+    DruidSegmentRecordReader testReader;
+    testReader = new DruidSegmentRecordReader(wikipedia_test, wikipedia_test_schema);
+
+    List<GenericRow> genericRows = new ArrayList<>();
+    while (testReader.hasNext()) {
+      GenericRow nextRow = testReader.next();
+      genericRows.add(nextRow);
+    }
+    Assert.assertEquals(genericRows.size(), wikipedia_size, "Generic row size must be " + wikipedia_size + ".");
+  }
+
+  @Test
+  public void testCSVRecord()
+      throws IOException {
+    final String[] columms = {"sum_INT_SV", "INT_MV", "__time"};
```

Comment:
`INT_MV` looks like `integer multivalue` column. Can you change this to `STRING_MV`? (String multivalue column is what you are actually testing)